src/Arkanis.Overlay.Host.Desktop/Workers/GameWindowTracker.cs

```diff
@@ -532,9 +491,95 @@ private bool GetWindowFocus(HWND? targetHWnd = null)
         return isFocused;
     }
 
+    private void UpdateWindowSizeAndPosition()
+    {
+        var windowPosition = GetWindowPosition();
+        var windowSize = GetWindowSize();
+
+        //? this is okay because it does by-value comparison internally
+        if (windowPosition != CurrentWindowPosition)
+        {
+            CurrentWindowPosition = windowPosition;
+            WindowPositionChanged?.Invoke(null, windowPosition);
+        }
+
+        //? this is okay because it does by-value comparison internally
+        if (windowSize != CurrentWindowSize)
+        {
+            CurrentWindowSize = windowSize;
+            WindowSizeChanged?.Invoke(null, windowSize);
+        }
+    }
+
+    #region Published Events
+
+    /// <summary>
+    ///     Raised when the window focus was changed.
+    ///     Reports true if the window is focused, false otherwise.
+    /// </summary>
+    public event EventHandler<bool>? WindowFocusChanged;
+
+    /// <summary>
+    ///     Raised when the tracked window changes.
+    ///     Reports true if there is a window active and being tracked, false otherwise.
+    /// </summary>
+    internal event EventHandler<WindowTrackingInfo>? WindowTrackingChanged;
+
+    /// <summary>
+    ///     Raised when the window position changed.
+    ///     Reports the new position of the window.
+    /// </summary>
+    public event EventHandler<Point>? WindowPositionChanged;
+
+    /// <summary>
+    ///     Raised when the window size changed.
+    ///     Reports the new size of the window.
+    /// </summary>
+    public event EventHandler<Size>? WindowSizeChanged;
+
+    /// <summary>
+    ///     Raised when the window size or position change has started.
+    ///     Use this event to hide the Overlay while the window is being moved or resized.
+    /// </summary>
+    public event EventHandler? WindowSizeOrPositionChangeStart;
+
+    /// <summary>
+    ///     Raised when the window size or position change has ended.
+    ///     Use this event to restore the Overlay after the window has finished being moved or resized.
+    /// </summary>
+    public event EventHandler? WindowSizeOrPositionChangeEnd;
+
+    #endregion
+
+    #region Internal Events
+
+    private event EventHandler? WindowMinimized;
+    private event EventHandler? WindowRestored;
+
+    /// <summary>
+    ///     Raised when the Star Citizen game window was found.
+    /// </summary>
+    private event EventHandler<HWND>? WindowFound;
+
+    /// <summary>
+    ///     Raised when the Star Citizen game window was lost.
+    ///     The game might have been closed or crashed.
+    /// </summary>
+    private event EventHandler? WindowLost;
+
+    /// <summary>
+    ///     Raised when the Star Citizen game process exits.
+    /// </summary>
+    private event EventHandler? ProcessExited;
+
+    #endregion
```

Comment:
Why not expose these directly? All of these were meant to be public, and for `WindowFound` / `WindowLost` it causes the problems explained on the struct `EmitWindowState` below.